frontend/test/pytest/test_compiler.py

```diff
@@ -303,7 +302,7 @@ def circuit():
         assert "While processing 'TestPass' pass of the 'PipelineB' pipeline" in e.value.args[0]
         assert "PipelineA" not in e.value.args[0]
         assert "Trace" not in e.value.args[0]
-        assert isfile(os.path.join(str(compiled.workspace), "2_PipelineB_FAILED.mlir"))
+        assert isfile(os.path.join(str(compiled.workspace), "3_1_PipelineB_FAILED.mlir"))
```

Comment:
Why are there two numbers now? E.g., `3_1` vs `2`?

Reply:
* The first number iterates over hand-coded stages that we have. We increase it manually, e.g. [here](https://github.com/PennyLaneAI/catalyst/blob/27443b0c79353043616852732c1e08b93f9603a7/mlir/lib/Driver/CompilerDriver.cpp#L412). Here we rely on the fact that compiler driver itself is not multi-threaded.
* The second number is the index of the pipeline in the list of pipelines provided by the user. These dumps might happen from different threads, but the file names will be different.

Reply:
I don't understand it fully, but is there a way to just enumerate them with a single number instead of this tuple? I don't think there is a difference between the pipelines provided by us and the user.

Reply:
It is possible of cause, but it would complicate the design with the arithmetic which I think is not necessary. We would need to add a number of pipelines to the counter at some point. But in reality we only want the filenames to be sorted.